src/all/exhentai/src/eu/kanade/tachiyomi/extension/all/exhentai/EXHentai.kt

```diff
@@ -0,0 +1,654 @@
+                } else {
+                    splitted.forEach { tag ->
+                        val trimmed = tag.trim().lowercase()
+                        if (trimmed.startsWith('-')) {
+                            modifiedQuery += " -${it.type}:\"${trimmed.removePrefix("-").replace(" ", "+")}\""
+                        } else {
+                            modifiedQuery += " ${it.type}:\"${trimmed.replace(" ", "+")}\""
+                        }
+                    }
+                }
+            }
+        }
+        uri.appendQueryParameter("f_search", modifiedQuery)
+        // when attempting to search with no genres selected, will auto select all genres
+        filters.filterIsInstance<GenreGroup>().firstOrNull()?.state?.let {
+            // variable to to check is any genres are selected
+            val check = it.any { option -> option.state } // or it.any(GenreOption::state)
+            // if no genres are selected by the user set all genres to on
+            if (!check) {
+                for (i in it) {
+                    i.state = true
+                }
+            }
+        }
+
+        filters.forEach {
+            if (it is UriFilter) it.addToUri(uri)
+        }
+
+        if (uri.toString().contains("f_spf") || uri.toString().contains("f_spt")) {
+            if (page > 1) uri.appendQueryParameter("from", lastMangaId)
+        }
+
+        return exGet(uri.toString(), page)
+    }
+
+    override fun latestUpdatesRequest(page: Int) = exGet(baseUrl, page)
+
+    override fun popularMangaParse(response: Response) = genericMangaParse(response)
+    override fun searchMangaParse(response: Response) = genericMangaParse(response)
+    override fun latestUpdatesParse(response: Response) = genericMangaParse(response)
+
+    private fun exGet(url: String, page: Int? = null, additionalHeaders: Headers? = null, cache: Boolean = true): Request {
+        // pages no longer exist, if app attempts to go to the first page after a request, do not include the page append
+        val pageIndex = if (page == 1) null else page
+        return GET(
+            pageIndex?.let {
+                addParam(url, "next", lastMangaId)
+            } ?: url,
+            additionalHeaders?.let { header ->
+                val headers = headers.newBuilder()
+                header.toMultimap().forEach { (t, u) ->
+                    u.forEach {
+                        headers.add(t, it)
+                    }
+                }
+                headers.build()
+            } ?: headers,
+
+        ).let {
+            if (!cache) {
+                it.newBuilder().cacheControl(CacheControl.FORCE_NETWORK).build()
+            } else {
+                it
+            }
+        }
+    }
+
+    /**
+     * Parse gallery page to metadata model
+     */
+    @SuppressLint("DefaultLocale")
+    override fun mangaDetailsParse(response: Response) = with(response.asJsoup()) {
```

Comment:
In `mangaDetailsParse()`: a lot of the `select` calls can be replaced with `selectFirst`, this change will also not silently hide if an element is missing.